### PR TITLE
Feature/docs stark-ui testing subpackage

### PR DESCRIPTION
**Pull request by christophercr**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1600

## What is the new behavior?
The docs of the `testing` subpackage from the Stark-UI are now generated and included in the Stark API Docs.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information
Took the chance to increase the coverage of the documentation 👍

### CODE REVIEW

File: packages/stark-ui/src/modules/table/components/table.component.ts [hunk 1]
```diff
@@ -219,13 +219,13 @@ export class StarkTableComponent extends AbstractStarkUiComponent implements OnI
...
 	/**
-	 * Shows or hides the minimap component in the header. When set to 'compact' it shows the compacted minimap
+	 * Shows or hides the {@link StarkMinimapComponent} in the header. When set to 'compact' it shows the compacted minimap
 	 */
 	@Input()
 	public minimap: boolean | StarkMinimapComponentMode = true;
 
 	/**
-	 * {@link StarkPaginationConfig} configuration object for embedded pagination component
+	 * {@link StarkPaginationConfig} object for the embedded {@link StarkPaginationComponent} component
```
[the remark this answers is not in the document]
Comment (christophercr) [reply packages/stark-ui/src/modules/table/components/table.component.ts]: Indeed... much better ;)

File: packages/stark-ui/testing/src/app-sidebar.mock.ts [hunk 2]
```diff
@@ -1,77 +1,50 @@
...
-		}).compileComponents();
-	}));
-
-});
-```
+ * Mock class of the {@link StarkAppSidebarService} interface.
  */
-/* tslint:enable:jsdoc-format */
-export class MockAppSidebarService implements SpyObj<StarkAppSidebarService> {
+export class MockStarkAppSidebarService implements SpyObj<StarkAppSidebarService> {
```
[the remark this answers is not in the document]
Comment (christophercr) [reply packages/stark-ui/testing/src/app-sidebar.mock.ts:10]: True, I didn't think of that. Now I've separated it in a different commit 👍

**Comment by christophercr** [comment]
@SuperITMan I've adapted the PR with the latest remarks, can you have a look?

**Comment by SuperITMan** [comment]
@christophercr Thanks for the modification.
About the additional commit, it does not appear in the CHANGELOG because you didn't mention the "BREAKING CHANGES", could you adapt it ?
for instance:
```
refactor(stark-ui): renamed `MockAppSidebarService` to `MockStarkAppSidebarService` in `@nationalbankbelgium/stark-ui/testing` subpackage

BREAKING CHANGES:
Renamed `MockAppSidebarService` to `MockStarkAppSidebarService` in `@nationalbankbelgium/stark-ui/testing` subpackage
```

I know it's redundant so any other idea would be better 😊 

Normally, `refactor(stark-ui)!: ...` would be enough to display the message in the CHANGELOG but it's not the case 😕
See: https://github.com/conventional-changelog/conventional-changelog/issues/648

**Comment by christophercr** [comment]
Mmm I'm no sure whether this is actually a breaking change. I mean, for the testing mocks... this is not necessarily a breaking change since the applications won't break. It will only break those unit tests where this mocks are used. So I would say we can leave it as a `fix` so it appears in the changelog. What do you think @SuperITMan ?

**Comment by SuperITMan** [comment]
For me, fix is ok 👌

**Comment by christophercr** [comment]
@SuperITMan I've adapted the commit message. 

Now the Travis build fails because of some linting errors in the starter... could be because of the new version of code-style. But never mind, I'm fixing these errors already in the PR for the upgrade of code-style :wink:

So I think this PRQ is ready to merge :+1:

**Comment by christophercr** [comment]
@SuperITMan I've rebased this PR and now it's ready to be merged 👍

**Comment by christophercr** [comment]
@SuperITMan I've rebased this PR and fixed all conflicts. It's ready to be merged :wink:

File: packages/stark-ui/testing/src/app-sidebar.mock.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        ...
+    ]
+});
+```
+
+Then you can just inject the Stark service via the TestBed using its corresponding `InjectionToken`:
+
+```typescript
+// this will inject the instantiated mock class
+mockStarkAppSidebarService = TetBed.get(STARK_APP_SIDEBAR_SERVICE);
```
Comment (SuperITMan) [inline packages/stark-ui/testing/src/app-sidebar.mock.md:27]: Small typo: Te**s**tBed.get

**Review by SuperITMan — CHANGES_REQUESTED** [review]
A small typo to fix and then we can merge this! 
Thanks again :+1:

File: packages/stark-ui/testing/src/app-sidebar.mock.md (hunk 3 above)
Comment (christophercr) [reply packages/stark-ui/testing/src/app-sidebar.mock.md]: Oops...

**Comment by christophercr** [comment]
@SuperITMan typos fixed 😉 
The other MD files had the same typo ...

**Review by SuperITMan — APPROVED** [review]
LGTM :+1:

**Comment by SuperITMan** [comment]
Oops, I didn't notice it for the other files... Thanks for the correction :blush:

**Pull request merged**